### PR TITLE
Refs #10308: add to_bool function for converting to boolean.

### PR DESCRIPTION
**Pull request by waldenraines**
http://projects.theforeman.org/issues/10308

### CODE REVIEW

**Comment by waldenraines** [comment]
See https://github.com/Katello/katello/pull/5198 for usage.

File: lib/cast.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Cast
+  def self.to_bool(value)
+    case value.class
```
Comment (daviddavis) [inline lib/cast.rb]: I think you just want value here and not `value.class`

File: lib/cast.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+      when String
+        return true if self == true || self =~ (/(true|t|yes|y|1)$/i)
```
Comment (daviddavis) [inline lib/cast.rb]: You use `self` everywhere but I think you mean `value`. Also, it's weird that this is checking if value is true if we know it's a String.

**Comment by daviddavis** [comment]
You could in theory use [ActiveRecord::ConnectionAdapters::Column.value_to_boolean](http://git.io/vJycS). It's a bit of a hack. Rails 4.2 has [ActiveRecord::Type::Boolean.new.cast_value](http://git.io/vJyC9).

File: lib/cast.rb (hunk 2 above)
Comment (waldenraines) [inline lib/cast.rb]: Hah, bad copy/paste, will fix.

**Comment by domcleal** [comment]
This code needs unit tests and to be used in the codebase.  There are probably a few places, like Setting and perhaps LookupValue, in which bool conversions are made already and it might be appropriate to change.  Perhaps also in some of our controllers?

Please file a ticket against Foreman and "fix" it, don't ref another project.

File: lib/cast.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        return false if value == 0
+
+      when NilClass
+        return false
+
+      else
+        return value
+    end
+  end
+end
```
Comment (domcleal) [inline lib/cast.rb:20]: missing trailing \n

File: lib/cast.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Cast
```
Comment (domcleal) [inline lib/cast.rb:1]: should be under Foreman's namespace

Comment (waldenraines) [inline lib/cast.rb:1]: Even though everything else in lib is not?

Comment (domcleal) [inline lib/cast.rb:1]: lib/foreman/ clearly is, everything else is a battle I have yet to start...

Comment (waldenraines) [inline lib/cast.rb:1]: Fair enough, shall I move this to the foreman directory as well as adding it to Foreman's namespace?

**Comment by waldenraines** [comment]
Closing in favor of https://github.com/theforeman/foreman/pull/2356 so the Foreman bot will sync up correctly with the newly created issue.

**Pull request closed without merging**